src/pages/index.js

```diff
@@ -0,0 +1,264 @@
+      });
+  }
+}
+
+function upDateAvatar(avatar) {
+  avatarPopup.setLoading(true);
+  api
+    .upDateAvater(avatar.link)
+    .then((res) => {
+      avatarImage.src = res.avatar;
+      avatarPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error in upDateAvatar:", error);
+      return Promise.reject(error);
+    })
+    .finally(() => {
+      avatarPopup.setLoading(false);
+    });
+}
+
+///////////////////////////////////////////////////////// Event Listeners /////////////////////////////////////////////////////////////////
+
+//////////////////////////// modal events ///////////////////
+// editButton.addEventListener("click", function () {
+//   api
+//     .getUserInfo()
+//     .then((userData) => {
+//       userInfo.getUserInfo(userData.name, userData.about);
+//       editFormValidator.resetValidation();
+//       nameInput.value = userData.name;
+//       jobInput.value = userData.about;
+//       newEditPopup.open();
+//     })
+//     .catch((error) => {
+//       console.error("Error in getUserInfo:", error);
+//       return Promise.reject(error);
+//     });
+// });
+
+editButton.addEventListener("click", function () {
+  nameInput.value = profileName.textContent;
+  jobInput.value = profileJob.textContent;
+  newEditPopup.open();
+});
+
+/////////// add modal events /////////////////////////
+addButton.addEventListener("click", () => {
+  addCardFormValidator.resetValidation();
+  newCardPopup.open();
+});
+
+updateAvatar.addEventListener("click", () => {
+  avatarUpdateValidator.resetValidation();
+  avatarPopup.open();
+});
+
+//////////////////// cards events //////////////////////
+
+editFormValidator.enableValidation(); //calling the enableValidation method from the new created var that has this method inside the class
+addCardFormValidator.enableValidation();
+avatarUpdateValidator.enableValidation();
+
+////////////////////API///////////////////////////////////
+api
+  .getInitialCards()
+  .then((cardData) => {
+    cardData.forEach((card) => {
+      const newCard = createCard(card);
+      sectionCards.addItem(newCard);
+    });
+    //by calling the api.getInitialCards method of this class and passing the renderItems method to the
+    //.then method if the get response is ok only then the renderItems will be called and the cards will
+    //be displyed to the page
+  })
+  .catch((error) => {
+    console.error("Error in getInitialCards:", error);
+    return Promise.reject(error);
+  });
+
+api.getUserInfo().then((userData) => {
+  userInfo.loadUserInfo(userData.name, userData.about, userData.avatar); //by calling the api.loadUserInfo method of this class
+  //and in the .then method after we get a response we calling the setUserInfo of the UserInfo class
+  //and passing it the userData as a propery{name,about and avatar} are proporty of the body we get from the
+  //API and in the setUserInfo we assinig the name and about to the name and description textContent
+});
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**

Reply:
fixed it